src/Microsoft.AspNetCore.Hosting.Abstractions/IApplicationLifetimeEvents.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Hosting
+{
+    /// <summary>
+    /// Allows consumers to perform cleanup during a graceful shutdown.
+    /// </summary>
+    public interface IApplicationLifetimeEvents
```

Comment:
Yep. The other problem with n interfaces today is the fact that our DI container doesn't let you do this:

services.AddSingleton<IApplicationLifetimeStarted, IApplicationLifetimeStopping, IApplicationLifetimeStopped, Thing>();